src/Command/ZeevouCreateAndSyncDoorCodesCommand.php

```
@@ -14,154 +14,384 @@ class ZeevouCreateAndSyncDoorCodesCommand extends Command
     * Post Request
     */
-    private function getRequest($uri)
+    private function postRequest($uri, $token, $postData)
     {
-        $client = HttpClient::create([
-            
+        $client = HttpClient::create();
+        $response = $client->request('POST', $uri, [
+                'json' => $postData,
+            'headers' => [
+                'Content-Type' => 'application/json',
+                'Authorization' => 'Bearer '.$token
+            ],
         ]);
-        $data = $this->getClientKeys($platform);
-        $response = $client->request('GET', $data['url'].$uri, [
+        $result['response_code'] = $response->getStatusCode();
+        $result['data'] = $response->toArray();
+        return $result;
+    }
+    
+    /**
+    * Put Request
+    */
+    private function putRequest($uri, $token, $postData)
+    {
+        $client = HttpClient::create();
+        $response = $client->request('PUT', $uri, [
+                'json' => $postData,
             'headers' => [
                 'Content-Type' => 'application/json',
+                'Authorization' => 'Bearer '.$token
             ],
         ]);
-        $content = $response->getContent();
+        $result['response_code'] = $response->getStatusCode();
+        $result['data'] = $response->toArray();
+        return $result;
+    }
+    
+    /**
+    * Delete Request
+    */
+    private function deleteRequest($uri, $token)
+    {
+        $client = HttpClient::create();
+        $response = $client->request('DELETE', $uri, [
+            'headers' => [
+                'Content-Type' => 'application/json',
+                'Authorization' => 'Bearer '.$token
+            ],
+        ]);
+        $result['response_code'] = $response->getStatusCode();
+        return $result;
+    }
+    
+    /**
+    * Write error and stop cron execution
+    * @param type $msg
+    */
+    private function writeErrorAndExit($output, $msg)
+    {
+        $output->writeln($msg);
+        die;
```

Why do you call `die` here? This will kill the php process and any clean-up code that the framework would execute is not executed. 
Rather throw an exception or error to allow proper error handling further up the stack (e.g. printing the output).

Check this page to read-up on how Exception handling is supposed to be done in PHP: https://www.w3schools.com/php/php_exception.asp

TL:DR: 
- if an error occurs that you cannot reasonably recover/rectify, then throw an `Exception`, maybe the caller know what to do
- if you write code that you think might cause an exception (and you know how to handle that case) then wrap that code in a `try-catch` and handle the error in the `catch`-section.
- if you catch an exception and know what to do to recover from the error, then do it. Else re-throw the exception or create a new exception with additional information.
- always throw the most specific type of `Exception` you can. E.g. file not found => throw FileNotFoundException if it is available in your framework.

---

Added try-catch throwing exception